sdks/python/apache_beam/ml/inference/vertex_ai_inference_it_test.py

```diff
@@ -45,6 +47,10 @@
 
 
 class VertexAIInference(unittest.TestCase):
+  @unittest.skipIf(
+      tf is None, 'Missing dependencies. '
```

Comment:
Given that we already have @pytest.mark.uses_vertex_ai , I would't do this skip. We might  accidentally silently disable the test and won't know it's not running.

Reply:
do we need to remove `@pytest.mark.it_postcommit` if this test is accidentally included into a suite where it is not supposed to run?

Reply:
actually, this is almost the same as statements in line 32. i guess i have a general question, how do we make sure this test is running in a suite where it is supposed to run in. It sounds like this test was passing before, and now it is failing, b/c TF containers are no longer installed, and the spirit of this change is to skip the test. 

sounds like something might be misconfigured. 
- If it was included in non-vertex ai postcommit - that's not expected - is it because we incorrectly included it_postcommit annotation? then we should remove that.
- If the test was included in a special vertex AI suite, but not postcommits, and now starts to fail  - i'd expect seeing changes pertaining to adding tensorflow in a requirements file, to make the test pass again.

Reply:
looks like we have https://github.com/apache/beam/blob/3915c85b84207f5f2c1e9dc4dd441769f3c586a5/sdks/python/apache_beam/ml/inference/vertex_ai_tests_requirements.txt#L19 , so probably need to remove it_postcommit annotation from this test to not include it twice.

Reply:
All of our other IT tests (tensorflow model handler IT test included) also have the it_postcommit annotation.

Reply:
We also *should* be getting the requirements file for that suite here - https://github.com/jrmccluskey/beam/blob/8f60924a3d5303d966ddaf80ee6ee7806edc346f/sdks/python/test-suites/dataflow/common.gradle#L408

Reply:
What's weird is that we're getting the gradle task invoked separately and not as part of the postcommit 

![image](https://github.com/apache/beam/assets/34928439/9ccd5b62-288b-4c75-9082-a1efd64058e0)

Reply:
ok, i see we 
require both markers in: https://github.com/apache/beam/blob/6f91ef83cade027523ea0a4d99c1edc2c41e7b3d/sdks/python/test-suites/dataflow/common.gradle#L420

Reply:
disregard that, it's WAI the limited load on the logs just obscured the rest of the suites listed

Reply:
> All of our other IT tests (tensorflow model handler IT test included) also have the it_postcommit annotation.

So, if we look at tests that have uses_tf, we have two kinds of tests: unit tests  and postcommit IT tests (the latter run a whole pipeline).
Since for vertex AI you don't have unit tests and only have postcommit tests, I think can simplify the markers in your suite and only use one marker uses_vertex_ai, and collect that test in postcommit suite.

Reply:
> What's weird is that we're getting the gradle task invoked separately and not as part of the postcommit

postCommitIT is a subset of all test that run in a postcommit Jenkins / GHA suite.  https://github.com/apache/beam/blob/a90879af543eede5bc7e99e383cfd666dab3ea59/build.gradle.kts#L497

Reply:
to clarify: vertex ai suite is a part of  inferencePostcommitIT suite:

https://github.com/apache/beam/blob/6c4bdf105567f5553fc738633e1d97af8bb63447/sdks/python/test-suites/dataflow/common.gradle#L470

Reply:
![image](https://github.com/apache/beam/assets/34928439/04b0a6a9-b939-4a47-b5dc-a6070e1300f5)

Tensorflow is getting installed and the test isn't getting skipped for lacking tensorflow, but the test is running on Dataflow where the worker doesn't have tensorflow - https://console.cloud.google.com/dataflow/jobs/us-central1/2023-09-19_10_21_28-17008252103869768778?project=apache-beam-testing